### PR TITLE
Fixes for issue 901 continued

### PR DESCRIPTION
**Pull request by dcousens**
@afk11 maybe this is what you intended in #909 ?

f4a83f8 is a follow up to #907 which was having issues locally

### CODE REVIEW

File: src/transaction_builder.js [hunk 1]
```diff
@@ -686,7 +682,13 @@ TransactionBuilder.prototype.sign = function (vin, keyPair, redeemScript, hashTy
 
   var kpPubKey = keyPair.getPublicKeyBuffer()
   if (!canSign(input)) {
-    prepareInput(input, kpPubKey, redeemScript, witnessValue, witnessScript)
+    if (witnessValue !== undefined) {
+      if (input.value !== undefined && input.value !== witnessValue) throw new Error('Input didn\'t match witnessValue')
+      typeforce(types.Satoshi, witnessValue)
+      input.value = witnessValue
+    }
```
Comment (dcousens) [inline src/transaction_builder.js]: @afk11

Comment (afk11) [reply src/transaction_builder.js]: I think this side steps the call to fixMultisigOrder, which would remove any signatures from the array which aren't valid?

Comment (dcousens) [reply src/transaction_builder.js]: @afk11 this isn't related to MS order,  it simply ensures that `.value` is added on if missing

**Comment by afk11** [comment]
@dcousens ah, I see the confusion.. I just found the exact same issue as mine recently: https://github.com/bitcoinjs/bitcoinjs-lib/issues/901#event-1266583387 which is indeed related to the input.value not being present, but the fixMsOrder demonstrates how it can be a problem if you leave it until `sign` - invalid signatures remain in the array unless fixMultisigOrder is called. 

So my solution was #909, which means that signatures fail and aren't extracted IF the transaction, which was passed to TransactionBuilder.fromTransaction as inputs which have the input.value for segwit outputs. 

Ping @dakk since we are having identical issues.

**Comment by dcousens** [comment]
As discussed on Slack,  we'll need to incorporate the second half of the complete fix for this problem in #909

**Comment by dakk** [comment]
I confirm it works now with my test case, excellent work

**Pull request merged**